doc/mgmt/SONiC Management Framework Show Techsupport HLD.md

```diff
@@ -0,0 +1,189 @@
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+The following Sonic Yang model is used for implementation of this feature:
+
+module: sonic-show-techsupport
+
+  rpcs:
+    +---x sonic-show-techsupport-info
+       +---w input
+       |  +---w date?   yang:date-and-time
+       +--ro output
+          +--ro output-filename?   string
+
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+N/A
+#### 3.6.2.2 Show Commands
+
+show techsupport [since <DateTime>] — Gather information for troubleshooting. Display the name of a file containing the resulting group of collected information items in a compressed "tar" file.
+
+Syntax Description:
+
+|    Keyword    | Description |
+|:-----------------|:-----------|
+| since DateTime | This option uses a text string (in the format returned by the Linux "date" command) to restrict the time scope for some of the information items to be collected (e.g. log files). It is passed to the host process and, if the date/time specification is valid, it is used during invocation of the applicable information gathering sub-functions.
+
+Command Mode: User EXEC
+
+Example:
+
+sonic# show techsupport
+
+/var/dump/sonic_dump_sonic_20191008_082312.tar.gz
+
+--------------------
+
+
+#### 3.6.2.3 Debug Commands
+N/A
+#### 3.6.2.4 IS-CLI Compliance
+N/A (TBD)
+
+The following table maps SONIC CLI commands to corresponding IS-CLI commands. The compliance column identifies how the command comply to the IS-CLI syntax:
+
+- **IS-CLI drop-in replace**  – meaning that it follows exactly the format of a pre-existing IS-CLI command.
+- **IS-CLI-like**  – meaning that the exact format of the IS-CLI command could not be followed, but the command is similar to other commands for IS-CLI (e.g. IS-CLI may not offer the exact option, but the command can be positioned is a similar manner as others for the related feature).
+- **SONIC** - meaning that no IS-CLI-like command could be found, so the command is derived specifically for SONIC.
+
+|CLI Command|Compliance|IS-CLI Command (if applicable)| Link to the web site identifying the IS-CLI command (if applicable)|
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| | | | |
+| | | | |
+| | | | |
+| | | | |
+| | | | |
+| | | | |
+| | | | |
+
+**Deviations from IS-CLI:** If there is a deviation from IS-CLI, Please state the reason(s).
+
+### 3.6.3 REST API Support
+TBD: (The base for RPC based Sonic Yang models is currently being developed.)
+
+# 4 Flow Diagrams
+
+# 5 Error Handling
+N/A
+
+# 6 Serviceability and Debug
+Logging, counters, stats, trace considerations. Please make sure you have incorporated the debugging framework feature. e.g., ensure your code registers with the debugging framework and add your dump routines for any debug info you want to be collected.
```

Comment:
Please replace the default text with content specific to your feature, or mark it as N/A

Reply:
Done. (Please see the new revision.)